src/licenses/LicenseModel.cpp

```diff
@@ -0,0 +1,184 @@
+#include "LicenseModel.h"
+
+#include <QDirIterator>
+
+static const char *const platformRootPath = ":/licenses";
+static const char *const qtRootPath = ":/licenses/qt";
+static const char *const toolkitRootPath = ":/licenses/qt/licenses";
+static const char *const webviewRootPath = ":/licenses/qt/licenses/qtwebengine/src/3rdparty/chromium";
+
+namespace licenses {
+
+  LicenseModel::LicenseModel(QObject *parent) : QAbstractItemModel(parent), m_dir(platformRootPath) {
+    populate();
+  }
+
+  LicenseModel::~LicenseModel() = default;
```

Comment:
Hm, with my rewrite, where I forward declare LicenseItem but use the type in my unique_ptr the compiler here on windows does not play well with having it in the header

Reply:
Hmmm.

I think unique_ptr needs to see the definition and not just a forward declaration, otherwise it doesn't get the deleter right.

But I'm not sure.  `¯\_(ツ)_/¯`